lib/rules/no-unnecessary-curly-literals.js

```diff
@@ -0,0 +1,24 @@
+import Rule from './_base.js';
+
+const LITERALS = new Set([
+  'StringLiteral',
+  'BooleanLiteral',
+  'NumberLiteral',
+  'UndefinedLiteral',
+  'NullLiteral',
+]);
+
+export default class NoUnnecessaryCurlyLiterals extends Rule {
+  visitor() {
+    return {
+      MustacheStatement(node) {
+        if (LITERALS.has(node.path.type)) {
+          this.log({
```

Comment:
We should keep the autofix for strings. We should only disable the autofix in the specific cases you're having issues with.

Reply:
even with strings i have the issue, say i have an example like this `< FooBar class={{'btn'}} />` this fix will change it to `< FooBar class="btn"}} />`. single quotes to double quotes. is that okay?

Reply:
That's fair. Let's preserve the type of quotes. Can you only enable the autofix when double quotes are present then? You can check the quoteType from the StringLiteral.

Reply:
sorry looks like i had blind spot, it didn't work for double quotes either(asserted wrong in my past commit where i check auto fix for double quotes). verified now. 😞

Reply:
Got it. Okay let's follow-up to autofix it.